src/component/vocabulary/RemoveVocabularyDialog.tsx

```diff
@@ -0,0 +1,239 @@
+import { ThunkDispatch } from "../../util/Types";
+import RDFStatement from "../../model/RDFStatement";
+import { trackPromise } from "react-promise-tracker";
+import "./RemoveVocabularyDialog.scss";
+import AssetLabel from "../misc/AssetLabel";
+import CopyIriIcon from "../misc/CopyIriIcon";
+import TermIriLink from "../term/TermIriLink";
+import VocabularyIriLink from "./VocabularyIriLink";
+import CustomInput from "../misc/CustomInput";
+import * as React from "react";
+import ValidationResult from "../../model/form/ValidationResult";
+import If from "../misc/If";
+
+interface RemoveVocabularyDialogProps {
+  show: boolean;
+  onSubmit: () => void;
+  onCancel: () => void;
+  vocabulary: Vocabulary;
+}
+
+const TRACKING_PROMISE_AREA = "vocabulary-remove-dialog";
+const b = (chunks: any) => <b>{chunks}</b>;
+
+const RemoveVocabularyDialog: React.FC<RemoveVocabularyDialogProps> = (
+  props
+) => {
+  const { i18n, formatMessage, locale } = useI18n();
+  const dispatch: ThunkDispatch = useDispatch();
+  const typeLabelId = Utils.getAssetTypeLabelId(props.vocabulary);
+  const typeLabel = i18n(typeLabelId || "type.asset").toLowerCase();
+
+  const [vocabularyRelations, setVocabularyRelations] = useState<
+    RDFStatement[]
+  >([]);
+  const [termsRelations, setTermsRelations] = useState<RDFStatement[]>([]);
+  const [confirmationVocabularyName, setconfirmationVocabularyName] =
+    useState("");
+
+  const canRemoveSilently = props.vocabulary.termCount === 0;
+
+  const canRemove =
+    vocabularyRelations.length === 0 && termsRelations.length === 0;
+
+  useEffect(() => {
+    if (!props.show) {
+      return;
+    }
+    setVocabularyRelations([]);
+    setTermsRelations([]);
+    setconfirmationVocabularyName("");
+
+    const iri = VocabularyUtils.create(props.vocabulary.iri);
+    const controller = new AbortController();
+
+    const vocabularyPromise = dispatch(
+      getVocabularyRelations(iri, controller)
+    ).then(setVocabularyRelations);
+
+    const termsPromise = dispatch(
+      getVocabularyTermsRelations(iri, controller)
+    ).then(setTermsRelations);
+
+    // track both promises
+    trackPromise(
+      Promise.all([vocabularyPromise, termsPromise]),
+      TRACKING_PROMISE_AREA
+    );
+
+    return () => controller.abort();
+  }, [props.vocabulary, props.show, dispatch]);
+
+  const cancel = () => {
+    props.onCancel();
+  };
+
+  const submit = () => {
+    if (validateInput() !== ValidationResult.VALID && !canRemoveSilently) {
+      return;
+    }
+    props.onSubmit();
+  };
+
+  const renderStatementArray = (
+    statements: RDFStatement[],
+    keyBase: string,
+    LinkComponent: typeof VocabularyIriLink | typeof TermIriLink
+  ) =>
+    statements.map((statement, index) => {
+      return (
+        <li key={keyBase + index}>
+          <span id={keyBase + index + "object"}>
+            <LinkComponent iri={statement.object.iri} />
+          </span>
+          <br />
```

Comment:
I don't know about this visualization. Wouldn't it be more readable to show the statement in a single line (with some spaces or a separator between the individual items) instead of three lines?

Reply:
IMHO Much better!